File: bindings/lua/Polycode/SceneManager.lua
class "SceneManager"


function SceneManager:SceneManager(...)
	local arg = {...}
	for k,v in pairs(arg) do
		if type(v) == "table" then
			if v.__ptr ~= nil then
				arg[k] = v.__ptr
			end
		end
	end
	if self.__ptr == nil and arg[1] ~= "__skip_ptr__" then
		self.__ptr = Polycode.SceneManager(unpack(arg))
	end
end

function SceneManager:Update()
	local retVal =  Polycode.SceneManager_Update(self.__ptr)
end

function SceneManager:fixedUpdate()
	local retVal =  Polycode.SceneManager_fixedUpdate(self.__ptr)
end

function SceneManager:Render(viewport)
	local retVal = Polycode.SceneManager_Render(self.__ptr, viewport.__ptr)
end

function SceneManager:renderVirtual()
	local retVal =  Polycode.SceneManager_renderVirtual(self.__ptr)
end

function SceneManager:__delete()
	if self then Polycode.delete_SceneManager(self.__ptr) end
end

File: bindings/javascript/Polycode/SceneManager.js
function SceneManager() {
	if(arguments[0] != "__skip_ptr__") {
		this.__ptr = Polycode.SceneManager()
	}
}


Duktape.fin(SceneManager.prototype, function (x) {
	if (x === SceneManager.prototype) {
		return;
	}
	Polycode.SceneManager__delete(x.__ptr)
})

SceneManager.prototype.Update = function() {
	Polycode.SceneManager_Update(this.__ptr)
}

SceneManager.prototype.fixedUpdate = function() {
	Polycode.SceneManager_fixedUpdate(this.__ptr)
}

SceneManager.prototype.Render = function(viewport) {
	Polycode.SceneManager_Render(this.__ptr, viewport)
}

SceneManager.prototype.renderVirtual = function() {
	Polycode.SceneManager_renderVirtual(this.__ptr)
}
